### PR TITLE
refs 22_04_implement_video_player_done

### PR DESCRIPTION
**Pull request by TheVinhLuong**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -69,6 +68,11 @@
...
             android:value="f70b5cb9235d9b64c8a0979b0380433f927aa1b4"/>
+
+        <activity
+            android:name=".screen.videoDetail.VideoDetailActivity"
+            android:screenOrientation="portrait"
+            android:theme="@style/AppTheme"/>
     </application>
 
-</manifest>
+</manifest>
```
Comment (daolq3012) [inline app/src/main/AndroidManifest.xml]: we must add blank line when end of file. so pls keep old code

File: app/src/main/java/com/framgia/englishconversation/screen/createPost/CreatePostViewModel.java [hunk 2]
```diff
@@ -220,6 +221,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 break;
             case REQUEST_RECORD_VIDEO:
                 updateVideoMedia(data.getData());
+
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/englishconversation/screen/createPost/CreatePostViewModel.java]: remove blank line

**Review by daolq3012 — CHANGES_REQUESTED** [review]
Pls check my comment!

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailPresenter.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.framgia.englishconversation.screen.videoDetail;
+
+/**
+ * Listens to user actions from the UI ({@link VideoDetailActivity}), retrieves the data and updates
+ * the UI as required.
+ */
+final class VideoDetailPresenter implements VideoDetailContract.Presenter {
+    private static final String TAG = VideoDetailPresenter.class.getName();
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailPresenter.java]: remove TAG

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    VideoDetailViewModel(Context context, TimelineModel timelineModel) {
+        mContext = context;
+        mTimelineModel = timelineModel;
+    }
+
+    @Override
+    public void onStart() {
+        mPresenter.onStart();
+        if (Util.SDK_INT > 23) {
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: Hard code. Add SDK_INT to Constant.java file

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            initializePlayer();
+        }
+    }
+
+    @Override
+    public void onResume() {
+        mPresenter.onResume();
+        if ((Util.SDK_INT <= 23) || mSimpleExoPlayer == null) {
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: Hard code

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            initializePlayer();
+        }
+    }
+
+    @Override
+    public void onPause() {
+        mPresenter.onPause();
+        if (Util.SDK_INT <= 23) {
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: Hard code

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            releasePlayer();
+        }
+    }
+
+    @Override
+    public void onStop() {
+        mPresenter.onStop();
+        if (Util.SDK_INT > 23) {
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: Hard code

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        notifyPropertyChanged(BR.simpleExoPlayer);
+    }
+
+    private void initializePlayer() {
+        SimpleExoPlayer player =
+                ExoPlayerFactory.newSimpleInstance(mContext, new DefaultTrackSelector());
+        setSimpleExoPlayer(player);
+        player.setPlayWhenReady(true);
+        player.seekTo(mCurrentPlaybackPosition);
+        Uri uri = Uri.parse(mTimelineModel.getMedias().get(0).getUrl());
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: Hard code

File: app/src/main/java/com/framgia/englishconversation/utils/Constant.java [hunk 9]
```diff
@@ -1,13 +1,12 @@
...
 
-import java.util.concurrent.TimeUnit;
-
 /**
  *
  */
 
 public class Constant {
     public static final String END_POINT_URL = "https://api.github.com";
+
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/utils/Constant.java]: remove blank line

File: app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java (hunk 8 above)
Comment (TheVinhLuong) [reply app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: i can't see where

Comment (daolq3012) [reply app/src/main/java/com/framgia/englishconversation/screen/videoDetail/VideoDetailViewModel.java]: check !=null in here

File: app/src/main/res/layout/custom_video_playback_controller.xml [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+    <LinearLayout
+        android:id="@+id/layout_linear_basic_control"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="center"
+        android:layout_centerInParent="true"
+        android:orientation="horizontal"
+        >
+
```
Comment (DoanVanToan) [inline app/src/main/res/layout/custom_video_playback_controller.xml]: remove bl

File: app/src/main/res/layout/custom_video_playback_controller.xml [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+                android:id="@id/exo_pause"
+                style="@style/ExoMediaButton.Pause"
+                />
+        </FrameLayout>
+
+        <ImageButton
+            android:id="@id/exo_next"
+            style="@style/ExoMediaButton.Next"
+            />
+
```
Comment (DoanVanToan) [inline app/src/main/res/layout/custom_video_playback_controller.xml:44]: remove bl

File: app/src/main/res/layout/custom_video_playback_controller.xml [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        >
+
+        <TextView
+            android:id="@id/exo_position"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:includeFontPadding="false"
+            android:paddingLeft="@dimen/dp_4"
+            android:paddingRight="@dimen/dp_4"
+            android:textColor="#FFBEBEBE"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/custom_video_playback_controller.xml]: hardcode

File: app/src/main/res/layout/custom_video_playback_controller.xml [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+            />
+
+        <TextView
+            android:id="@id/exo_duration"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:includeFontPadding="false"
+            android:paddingLeft="@dimen/dp_4"
+            android:paddingRight="@dimen/dp_4"
+            android:textColor="#FFBEBEBE"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/custom_video_playback_controller.xml]: hardcode

**Review by DoanVanToan — APPROVED** [review]

**Pull request merged**